molgenis-app-manager/src/main/java/org/molgenis/app/manager/service/impl/AppManagerServiceImpl.java

```diff
@@ -0,0 +1,261 @@
+package org.molgenis.app.manager.service.impl;
+
+import com.google.common.collect.Maps;
+import com.google.gson.Gson;
+import net.lingala.zip4j.core.ZipFile;
+import net.lingala.zip4j.exception.ZipException;
+import org.molgenis.app.manager.exception.*;
+import org.molgenis.app.manager.meta.App;
+import org.molgenis.app.manager.meta.AppFactory;
+import org.molgenis.app.manager.meta.AppMetadata;
+import org.molgenis.app.manager.model.AppConfig;
+import org.molgenis.app.manager.model.AppResponse;
+import org.molgenis.app.manager.service.AppManagerService;
+import org.molgenis.data.DataService;
+import org.molgenis.data.Query;
+import org.molgenis.data.file.FileStore;
+import org.molgenis.data.plugin.model.Plugin;
+import org.molgenis.data.plugin.model.PluginFactory;
+import org.molgenis.data.plugin.model.PluginMetadata;
+import org.molgenis.data.support.QueryImpl;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.io.FileUtils.deleteDirectory;
+
+@Service
+public class AppManagerServiceImpl implements AppManagerService
```

Comment:
Test coverage is to low for this class, 'uploadApp' and private helper methods are not covered by unit tests

Reply:
PR